src/main/scala/tech/cryptonomic/nautilus/metering/repositories/InfluxDbRepo.scala

```diff
@@ -52,15 +52,19 @@ class InfluxDbRepoImpl(cfg: InfluxDbConfig)(
 ) extends InfluxDbRepo
     with LazyLogging {
 
-  val cred = InfluxCredentials(cfg.username, cfg.password)
-  val client = new AkkaIOClient(cfg.host, cfg.port, Some(cred), false, None, false)
+  private val cred = InfluxCredentials(cfg.username, cfg.password)
+  private val client = new AkkaIOClient(cfg.host, cfg.port, Some(cred), false, None, false)
+
+  private val timeConstraint: Option[Long] => String = _.map(from => s" and time > $from").getOrElse("")
 
   /** Reads amount of queries grouped by 5minute intervals */
-  override def get5minQueries(apiKeys: List[String]): Future[List[QueryCQ]] =
+  override def get5minQueries(apiKeys: List[String], from: Option[Long]): Future[List[QueryCQ]] =
     client
       .measurement[QueryCQ](cfg.database, "five_minute_queries_measurement")
       .read(
-        s"select time, apiKey, count from a_day.five_minute_queries_measurement where apiKey =~ /${apiKeys.mkString("|")}/"
+        s"select time, apiKey, count from a_day.five_minute_queries_measurement where apiKey =~ /${apiKeys.mkString("|")}/" + timeConstraint(
+              from
```

Comment:
Is this appending a string fragment to an interpolation? Why not interpolate this value as well?

Reply:
ok, changed